pkg/sdk/storage_tus.go

```diff
@@ -0,0 +1,161 @@
+package sdk
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"connectrpc.com/connect"
+	v1storage "github.com/OpenAudio/go-openaudio/pkg/api/storage/v1"
+	storagev1connect "github.com/OpenAudio/go-openaudio/pkg/api/storage/v1/v1connect"
+	"github.com/bdragon300/tusgo"
+)
+
+type StorageServiceClient interface {
+	storagev1connect.StorageServiceClient
+	UploadFilesTus(context.Context, *connect.Request[v1storage.UploadFilesRequest]) (*connect.Response[v1storage.UploadFilesResponse], error)
+}
+
+type StorageServiceClientWithTUS struct {
+	storagev1connect.StorageServiceClient
+	sdk *OpenAudioSDK
+}
+
+// UploadFilesTus implements StorageServiceClient.UploadFilesTus.
+// It uploads files using the TUS resumable upload protocol.
+func (s *StorageServiceClientWithTUS) UploadFilesTus(ctx context.Context, req *connect.Request[v1storage.UploadFilesRequest]) (*connect.Response[v1storage.UploadFilesResponse], error) {
+	if len(req.Msg.Files) == 0 {
+		return nil, fmt.Errorf("no files provided")
+	}
+
+	if len(req.Msg.Files) > 1 {
+		return nil, fmt.Errorf("multiple files not supported")
+	}
+
+	file := req.Msg.Files[0]
+	if file.Filename == "" {
+		return nil, fmt.Errorf("filename is required")
+	}
+
+	tmpFile, err := os.CreateTemp("", "tus-upload-*")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temp file: %w", err)
+	}
+	defer os.Remove(tmpFile.Name())
+	defer tmpFile.Close()
+
+	if _, err := tmpFile.Write(file.Data); err != nil {
+		return nil, fmt.Errorf("failed to write to temp file: %w", err)
+	}
+
+	if _, err := tmpFile.Seek(0, io.SeekStart); err != nil {
+		return nil, fmt.Errorf("failed to seek temp file: %w", err)
+	}
+
+	fileInfo, err := tmpFile.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get file info: %w", err)
+	}
+	fileSize := fileInfo.Size()
+
+	serverURL := s.sdk.getServerURL()
+	if serverURL == "" {
+		return nil, fmt.Errorf("server URL not available")
+	}
+
+	tusBaseURL, err := url.Parse(fmt.Sprintf("%s/files/", serverURL))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse TUS base URL: %w", err)
+	}
+	tusClient := tusgo.NewClient(http.DefaultClient, tusBaseURL)
```

Comment:
seems weird to create a new client each time, but guess it's fine?

Reply:
do you think instead it would make sense to initialize the struct with the tus client instead? it would remove the circular dep